### PR TITLE
fix: complete path of existing record expr

### PR DESCRIPTION
**Pull request by cynecx**

### CODE REVIEW

File: crates/ide-completion/src/context/analysis.rs [hunk 1]
```diff
@@ -943,6 +943,7 @@ impl<'a> CompletionContext<'a> {
                         }
                     }
                 },
+                ast::RecordExpr(it) => make_path_kind_expr(it.into()),
```
Comment (Veykril) [inline crates/ide-completion/src/context/analysis.rs:946]: This line also needs to go into the match in the `// completing inside a qualifier` annotated part as well

File: crates/ide-completion/src/tests/expression.rs [hunk 2]
```diff
@@ -651,3 +651,22 @@ fn main() {
...
+impl Zulu {
+    fn test() -> Self { }
+}
+fn boi(val: Zulu) { }
+fn main() {
+    boi(Zulu:: $0 {});
+}
+"#,
+        expect![[r#"
+            fn test() fn() -> Zulu
```
Comment (Veykril) [inline crates/ide-completion/src/tests/expression.rs:669]: Can you add a FIXME here, we shouldn't be completing functions in this position, only structs, enum variants and unions

File: crates/parser/src/grammar/paths.rs [hunk 3]
```diff
@@ -54,7 +54,7 @@ fn path_for_qualifier(
     mut qual: CompletedMarker,
 ) -> CompletedMarker {
     loop {
-        let use_tree = matches!(p.nth(2), T![*] | T!['{']);
+        let use_tree = mode == Mode::Use && matches!(p.nth(2), T![*] | T!['{']);
```
Comment (Veykril) [inline crates/parser/src/grammar/paths.rs:57]: What is the purpose of this change?

Comment (cynecx) [reply crates/parser/src/grammar/paths.rs:57]: This improves the ast in case of a syntax error and recovers better imo.

| Before | After |
| --- | --- |
| image | image |

I initially thought that this was required so completions can work but later on I found out that completions actually "injects `intellijRulezz` into ast". So this better parsing recovery isn't actually needed but I thought it would be an overall improvement. But I can also leave it out, I don't mind either way.

File: crates/ide-completion/src/tests/expression.rs (hunk 2 above)
Comment (cynecx) [reply crates/ide-completion/src/tests/expression.rs:669]: Actually this was the reason for opening this PR in the first place. I initially had code like:

```rust
fn main() {
    boi(Zulu { ... });
}
```

But I wanted to search for a method on `Zulu` instead, so instead of actually deleting the `{ ... }`, I started typing typing `Zulu::<|>`. Before this PR this wouldn't have worked.

File: crates/parser/src/grammar/paths.rs (hunk 3 above)
Comment (Veykril) [reply crates/parser/src/grammar/paths.rs:57]: Aight, this change seems fine to me, and I do think that's actually needed to work for this here looking at the parse trees

File: crates/ide-completion/src/tests/expression.rs (hunk 2 above)
Comment (Veykril) [reply crates/ide-completion/src/tests/expression.rs:669]: Mmh, ye I think this might be us doing too much if we get rid of those completions, you are right. We should probably look into prioritizing records though, not relevant here now though 👍
